packages/loader/container-loader/src/connectionManager.ts

```diff
@@ -567,7 +574,9 @@ export class ConnectionManager implements IConnectionManager {
 					origError !== null &&
 					origError?.errorType === DriverErrorType.deltaStreamConnectionForbidden
 				) {
-					connection = new NoDeltaStream();
+					connection = new NoDeltaStream(
+						(origError as DeltaStreamConnectionForbiddenError).storageOnlyReason,
```

Comment:
the avoid as unsafe `as` cast, can we take the above if check, and put it in a typeguard function that narrows the type.